File: problems/2026/03/2026-03-27-two-sum/SolutionLC1.java
import java.util.HashMap;
import java.util.Map;

public class SolutionLC1 {

    public int[] twoSum(int[] nums, int target) {
        // 哈希表：记录已经遍历过的数字和它的下标
        Map<Integer, Integer> map = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            // 目标值减当前值，得到当前数字想要的“搭档”
            int need = target - nums[i];

            // 如果搭档已经出现过，直接返回搭档下标 + 当前下标
            if (map.containsKey(need)) {
                return new int[]{map.get(need), i};
            }

            // 否则先记录当前数字，给后续元素做搭档
            map.put(nums[i], i);
        }

        // 题目通常保证有解，这里是保底返回
        return new int[0];
    }
}

File: problems/2026/03/2026-03-27-two-sum/SolutionLC1.py
from typing import List


class Solution:
    def two_sum(self, nums: List[int], target: int) -> List[int]:
        # 哈希表：记录“数字 -> 下标”
        seen = {}

        for i, num in enumerate(nums):
            # 当前数字需要的搭档
            need = target - num

            # 找到搭档则直接返回答案
            if need in seen:
                return [seen[need], i]

            # 记录当前数字，供后续匹配
            seen[num] = i

        # 保底返回（按题意通常不会触发）
        return []
